megamek/src/megamek/server/Server.java

```diff
@@ -887,6 +887,16 @@ private void receivePlayerVersion(Packet packet, int connId) {
             System.out.println("ERROR: Client/Server Version Mismatch -- Client: "+version+" Server: "+MegaMek.VERSION);
             needs = true;
         }
+        else if (clientChecksum == null) {    // print a message indicating client doesn't have jar file
+            buf.append("Client Checksum is null. Client may not have a jar file");
+            System.out.println("ERROR: Client does not have a jar file");
+            needs = true;      	
+        }
+        else if (serverChecksum == null) {    // print message indicating server doesn't have jar file
+            buf.append("Server Checksum is null. Server may not have a jar file");
+            System.out.println("ERROR: Server does not have a jar file");
+            needs = true; 
+        }
         if (!clientChecksum.equals(serverChecksum)) {
```

Comment:
Hah, missed that one.  I had it as an else-if in my dropship branch and copied it over wrong.  Anything else I should know about the coding conventions?  I am trying to use 4 spaces instead of tab to indent, based on a convention I remember from my 2010 work.  I'm trying to put braces inline with their statements, though I know I missed a bunch of those in my dropship branch before I got the word.